compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -2128,6 +2165,29 @@ void OMR::X86::CodeGenerator::doBinaryEncoding()
          }
 
       self()->addToAtlas(cursorInstruction);
+
+      // If this is the last warm instruction, save info about the warm code range
+      // and set up to generate code in the cold code range.
+      //
+      if (cursorInstruction->isLastWarmInstruction())
+         {
+         setWarmCodeEnd(getBinaryBufferCursor());
+         setColdCodeStart(coldCode);
+         setBinaryBufferCursor(coldCode);
+
+         if (self()->comp()->getOption(TR_TraceCG))
+            {
+            traceMsg(self()->comp(), "%s warmCodeEnd = %p, lastWarmInstruction = %p coldCodeStart = %p\n",
+                                                           SPLIT_WARM_COLD_STRING,
+                                                           getWarmCodeEnd(), cursorInstruction, coldCode);
+            }
+
+         // Adjust the accumulated length error so that distances within the cold
+         // code are calculated properly using the estimated code locations.
+         //
+         setAccumulatedInstructionLengthError((int32_t)((self()->getBinaryBufferStart() + warmEstimate) + MIN_DISTANCE_BETWEEN_WARM_AND_COLD_CODE - coldCode));
```

Comment:
Should this maybe use `self()->getEstimatedWarmLength()` instead of `warmEstimate`, since above the estimate was set to `warmEstimate + OVER_ESTIMATION`?

Reply:
I think you are right. Fixed.